### DISCUSSION TITLE
Consulta sobre entityManager.persist(...)

### DISCUSSION THREAD

**Discussion by AlejandroDeheza** [dds-jv/foro#326] [Consultas de persistencia] [answered]
Buenas, viendo el código que vimos el viernes pasado me surgió una duda:

en [RepositorioUsuarios.java](https://github.com/dds-utn/jpa-proof-of-concept-template/blob/1a0d3b422c6669b591c8c6924e3f17074525fd38/src/main/java/model/RepositorioUsuarios.java#L26-L28)
```java
  public void agregar(Usuario usuario) {
    entityManager().persist(usuario);
  }
```

y en [RepositorioConsultoras.java](https://github.com/dds-utn/jpa-proof-of-concept-template/blob/1a0d3b422c6669b591c8c6924e3f17074525fd38/src/main/java/model/RepositorioConsultoras.java#L27-L29)
```java
  public void agregar(Consultora consultora) {
    entityManager().persist(consultora);
  }
```

En ambos metodos se hace lo mismo, solo que difieren en el tipado del parámetro. 

entonces, ¿Por qué no simplemente llamar a 
```java
  entityManager().persist(instanciaPersistible);
  ...
  entityManager().persist(otraInstanciaPersistible);
```
en vez de 
```java
  unRepo.agregar(instanciaPersistible);
  ...
  otroRepo.agregar(otraInstanciaPersistible);
```
?

Creería que solo se hace un pasamanos y que ese método se va a repetir en todos los repositorios que tengamos.

Creo que el método repetido en todos los Repositorios se podria arreglar con herencia, pero me sigue quedando la duda de si hacia falta en primer lugar.

Saludos!

**Comment by gastonprieto** [comment, marked as answer]
Buenas @AlejandroDeheza 

Te voy a plantear unas preguntas, que pasaba cuando no habiamos visto Hibernate? teniamos los repos? que pasa si dejamos de persistir usando Hibernate y usamos otro framework? o si usamos otro tipo de base de datos?

**Reply by AlejandroDeheza** [reply]
Hola @gastonprieto, creo que entiendo, los repositorios funcionarían como un adapter entre el modelo de objetos y la tecnología de persistencia que usemos, puede ser? Creo que también, envolviendo el metodo de ```entityManager().persist(algo)``` con ```repo.agregar(algo)```ganamos en declaratividad por si ese metodo es mas complejo en otro framework.

Eso seria mas o menos correcto?
Gracias!

**Reply by gastonprieto** [reply]
Es correcto lo que decís, lo único que no diria que ganas declaratividad (porque no ocultas detalles algoritmicos para darselos a un motor) sino que ganas en desacoplamiento o ocultas detalles de implementación. Pero si, esa es la idea
